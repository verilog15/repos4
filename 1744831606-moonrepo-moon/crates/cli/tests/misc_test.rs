use moon_test_utils::{
    create_sandbox_with_config, get_cases_fixture_configs, predicates::prelude::*,
};
use proto_core::VersionReq;

#[test]
fn fails_on_version_constraint() {
    let (mut workspace_config, _, _) = get_cases_fixture_configs();

    workspace_config.version_constraint = Some(VersionReq::parse(">=1000.0.0").unwrap());

    let sandbox = create_sandbox_with_config("cases", Some(workspace_config), None, None);

    let assert = sandbox.run_moon(|cmd| {
        cmd.arg("sync");
    });

    assert
        .failure()
        .stderr(predicate::str::contains(">=1000.0.0"));
}
