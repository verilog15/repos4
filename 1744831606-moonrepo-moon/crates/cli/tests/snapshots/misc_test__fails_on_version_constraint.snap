---
source: crates/cli/tests/misc_test.rs
expression: assert.output()
---

 ERROR 

Invalid moon version, unable to proceed. Found 0.22.0, expected >=1000.0.0.



